Mapsui.Geometries/ArrowLine.cs

```diff
@@ -0,0 +1,196 @@
+            }
+        }
+
+        /// <summary>
+        ///     the Point where this Geometry ends
+        /// </summary>
+        public Point EndPoint
+        {
+            get
+            {
+                return LineStrings[0].EndPoint;
+            }
+        }
+
+        /// <summary>
+        ///     The length of an arrow's side
+        /// </summary>
+        public double ArrowLength
+        {
+            get
+            {
+                return LineStrings[1].StartPoint.Distance(LineStrings[1].EndPoint);
+            }
+        }
+
+        /// <summary>
+        ///     The length of this ArrowLine, as measured in the spatial reference system of this ArrowLine.
+        /// </summary>
+        public new double Length
+        {
+            get
+            {
+                return StartPoint.Distance(EndPoint);
+            }
+        }
+
+        /// <summary>
+        ///     The position of the arrow on the line, placed on the center of the line by default 
+        /// </summary>
+        public Point ArrowHead
+        {
+            get
+            {
+                return new Point((StartPoint.X + EndPoint.X) / 2, (StartPoint.Y + EndPoint.Y) / 2);
+            }
+        }
+
+        /// <summary>
+        ///     The two points linked to the 'ArrowHead' in order to draw the sides of the arrow
+        /// </summary>
+        /// <param name="radiansAngle"></param>   
+        /// <returns>Returns a set of two Points</returns>
+        public Point[] ArrowExtremities(double radiansAngle, double length)
+        {
+            if (IsEmpty())
+                return new Point[] { ArrowHead, ArrowHead };
+
+            double slope;
+            if (StartPoint.X == EndPoint.X)  //Prevents div/0
+                slope = double.PositiveInfinity;
+            else
+                slope = (StartPoint.Y - EndPoint.Y) / (StartPoint.X - EndPoint.X);
+
+            double arrowOrientation = (StartPoint.X - EndPoint.X) / Math.Abs(StartPoint.X - EndPoint.X);
+
+            Point firstExt = new Point(ArrowHead.X + arrowOrientation * length * Math.Cos(Math.Atan(slope) - radiansAngle), ArrowHead.Y + arrowOrientation * length * Math.Sin(Math.Atan(slope) - radiansAngle));
+            Point secondExt = new Point(ArrowHead.X + arrowOrientation * length * Math.Cos(Math.Atan(slope) + radiansAngle), ArrowHead.Y + arrowOrientation * length * Math.Sin(Math.Atan(slope) + radiansAngle));
+
+            return new Point[] { firstExt, secondExt };
+        }
+
+        /// <summary>
+        ///     The minimum bounding box for this Geometry.
+        /// </summary>
+        /// <returns>BoundingBox for this geometry</returns>
+        public override BoundingBox GetBoundingBox()
+        {
+            if (IsEmpty())
+                return null;
+            Point[] pointsCollection = new Point[] { StartPoint, EndPoint, ArrowExtremities(RadiansAngle, ArrowLength)[0], ArrowExtremities(RadiansAngle, ArrowLength)[1] };
```

Comment:
This is inefficient, ArrowExtremities points should be known to the current object without need to recalculate every time the bounding box is calculated.